### PR TITLE
fix(Core/Creature): Move hardcoded text/gossips to DB - boss_xt002

### PR DESCRIPTION
**Pull request by avarishd**
Part of https://github.com/azerothcore/azerothcore-wotlk/projects/15 but xt was not on the list. (Going trough entirety of ulduar)
All of the creature_text is already in DB
However, SAY_SUMMON is not linked with BroadcastTextId and I cannot find any video source of it actually being used. (Except TC boss script, but still, no actual proof)
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes none

## Tests Performed:

- Tested in-game with sound
- 

## How to Test the Changes:

1. .go c 136054
2. Pull (with sound enabled)
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp [hunk 1]
```diff
@@ -91,17 +91,20 @@ enum NPCs
     NPC_PILE_TRIGGER            = 33337,
 };
 
-enum Sounds
+enum XT002Text
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp:94]:
```suggestion
enum Text
```

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp [hunk 2]
```diff
@@ -91,17 +91,20 @@ enum NPCs
...
+    SAY_HEART_CLOSED            = 2,
+    SAY_TYMPANIC_TANTRUM        = 3,
+    SAY_SLAY                    = 4,
+    SAY_BERSERK                 = 5,
+    SAY_DEATH                   = 6,
+    SAY_SUMMON                  = 7,
+    SAY_EMOTE_HEART_OPENED      = 8,
+    SAY_EMOTE_HEART_CLOSED      = 9,
+    SAY_EMOTE_TYMPANIC_TANTRUM  = 10,
+    SAY_EMOTE_SCRAPBOT          = 11,
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp:107]: should just be EMOTE_

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp (hunk 2 above)
Comment (avarishd) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp]: Done, good to know I guess.

**Review by Kitzunu — APPROVED** [review]

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp]: https://www.azerothcore.org/wiki/cpp-code-standards#standard-prefixes-for-constants if you want to learn more

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**